snap/pack/pack.go

```diff
@@ -107,27 +107,33 @@ func CheckSkeleton(w io.Writer, sourceDir string) error {
 }
 
 func loadAndValidate(sourceDir string) (*snap.Info, error) {
-	// ensure we have valid content
+	// Parsing of snap info is duplicated in ReadInfoFromSnapFile. It is done
+	// here to retrieve the snap instance name, if available, to use in case of
+	// ReadInfoFromSnapFile error
 	yaml, err := ioutil.ReadFile(filepath.Join(sourceDir, "meta", "snap.yaml"))
 	if err != nil {
 		return nil, err
 	}
-
 	info, err := snap.InfoFromSnapYaml(yaml)
 	if err != nil {
 		return nil, err
 	}
+	instanceName := info.InstanceName()
 
-	if err := snap.Validate(info); err != nil {
-		return nil, fmt.Errorf("cannot validate snap %q: %v", info.InstanceName(), err)
+	// ReadInfoFromSnapFile covers the following required steps:
+	// 	- Read snap metadata from meta/snap.yaml
+	//      - Parse snap info from meta/snap.yaml without side info
+	//      - Add and bind implicit hooks from meta/hooks
+	//      - Validate available snap information
+	//      - Validate snapshot metadata from opional meta/snapshot.yaml
+	info, err = snap.ReadInfoFromSnapFile(snapdir.New(sourceDir), nil)
```

Comment:
`ReadInfoFromSnapFile` covers everything we need including hook validation that was missing before. The hook validation itself was also extended to include default-configure with missing configure check [here](https://github.com/snapcore/snapd/pull/13097/files#diff-291f552d73b5fd8ca18103ba5b54a36155556275d87bc73f9160086fc5e347d1R165).